drivers/SmartThings/zwave-thermostat/src/ct100-thermostat/init.lua

```diff
@@ -117,6 +119,36 @@ local function thermostat_mode_report_handler(self, device, cmd)
   end
 end
 
+-- The context for this handler is that the ct100 has been observed to use the
+-- multiinstance command encap (0x06) and then pass in multi channel
+-- command encap arguments (i.e. including a destination endpoint).
+-- This causes everything to be off by 1 byte, and the dest endpoint to be
+-- parsed as the command class. Real nasty business. -sg
+
+-- e.g.:
+-- CC:Multi-Channel ID:0x06 Len:8 Payload:0x01 00 31 05 01 2A 02 58 Encap:None
+-- parsed as:
+-- (Thermostat)> received Z-Wave command: {args={command=49, command_class=0, instance=1, parameter="\x05\x01\x2A\x02\x58", res=false},
+-- cmd_class="MULTI_CHANNEL", cmd_id="MULTI_INSTANCE_CMD_ENCAP", dst_channels={}, encap="NONE", payload="\x01\x00\x31\x05\x01\x2A\x02\x58",
+-- src_channel=0, version=2}
+local function multi_instance_encap_handler(self, device, cmd)
+  if (cmd.args.command == cc.SENSOR_MULTILEVEL and
+    string.byte(cmd.args.parameter, 1, 2) == SensorMultilevel.REPORT) then
+    local size_scale_precision = string.byte(string.sub(cmd.args.parameter, 3, 4))
+    local precision = bit32.band(bit32.rshift(size_scale_precision, 5), 0x7) -- last 3 bits
+    local sensor_value = utils.bit_list_to_int(utils.bitify(string.sub(cmd.args.parameter, 4))) / (10 ^ precision)
+
+    local repack = SensorMultilevel:Report({
+      sensor_type = string.byte(string.sub(cmd.args.parameter, 2, 3)),
+      size = bit32.band(size_scale_precision, 0x7), -- first three bits
```

Comment:
someone check my math, is `bit32` available on the hub?

Reply:
yes

Reply:
everything I see online says it's part of the standard library but luacheck is mad?

Reply:
just putting an explicit `require` that should help

Reply:
for some reason I didn't think lua had actual bitwise operators